tests/integration/data/test_subset_with_crs.py

```diff
@@ -0,0 +1,102 @@
+import os
+
+import numpy as np
+import iris
+import shapely, shapely.geometry
+import cartopy.crs as ccrs
+
+from clean_air.data import DataSubset
+from clean_air import util
+
+SAMPLEDIR = os.path.expanduser("~cbosley/Projects/toybox/cap_sample_data")
+
+
+def test_latlon_point():
+    # Create example dataset
+    ds = DataSubset(
+        None,
+        "aqum",
+        os.path.join(SAMPLEDIR, "model", "aqum_daily*"),
+        point=(-0.1, 51.5),
+        crs=ccrs.Geodetic(),
+    )
+    cube = ds.as_cube()
+
+    xcoord, ycoord = util.cubes.get_xy_coords(cube)
+
+    # Check we have the point we asked for
+    assert iris.util.array_equal(xcoord.points.round(4), [531866.1304])
+    assert iris.util.array_equal(ycoord.points.round(4), [179660.9048])
+
+
+def test_latlon_box():
+    # Create example dataset
+    ds = DataSubset(
+        None,
+        "aqum",
+        os.path.join(SAMPLEDIR, "model", "aqum_daily*"),
+        box=(-4, 50.4, -2.8, 51.2),
+        crs=ccrs.Geodetic(),
+    )
+    cube = ds.as_cube()
+
+    xcoord, ycoord = util.cubes.get_xy_coords(cube)
+
+    # Check we have the points we asked for (multiples of 2000m within
+    # each range)
+    # Strictly speaking, the transformed box would have slightly curved edges,
+    # and the "corner-most" gridpoints would be:
+    # tl: (262000, 146000)
+    # tr: (344000, 144000)
+    # br: (342000, 56000)
+    # bl: (258000, 58000)
+    # We therefore expect a slightly larger box, covering the minimum and
+    # maximum in both directions
+    assert iris.util.array_equal(xcoord.points[[0, -1]], [258000, 344000])
+    assert iris.util.array_equal(ycoord.points[[0, -1]], [56000, 146000])
+
+
+class TestLatlonPolygonSubset:
+    # This is literally a copy of the unit test, with converted coordinates
+    def setup_class(self):
+        # Define a test polygon (an extremely simple representation of Exeter)
+        shape = shapely.geometry.Polygon([
+            (-3.5690, 50.7273),
+            (-3.5685, 50.7445),
+            (-3.5115, 50.7609),
+            (-3.4640, 50.7445),
+            (-3.4555, 50.6727),
+            (-3.4984, 50.6936),
+            (-3.5379, 50.6924),
+        ])
+
+        # Create example dataset
+        self.ds = DataSubset(
+            None,
+            "aqum",
+            os.path.join(SAMPLEDIR, "model", "aqum_hourly_o3_20200520.nc"),
```

Comment:
Yes, I did feel a need for at least a slightly larger dataset than currently in that repo, to be sure it is genuinely picking out the right area. See in particular the box test, which needed to be large enough that the corners were definitely in different rows and columns. I did a fair amount of plotting to verify the tests as well, which was a easier when it could include familiar coastlines, but not necessary.

I'd rather leave it as is for this PR, if that's OK, seeing as the only change would be the actual path to the file, which will all change anyway when we properly integrate the testing.